state/models.go

```diff
@@ -1221,3 +1229,55 @@ type RunTags struct {
 	TeamName   string `json:"team_name"`
 	OwnerID    string `json:"owner_id"`
 }
+
+type ClusterStatus string
+
+type Tier string
+type Tiers []Tier
+
+const (
+	Tier0 Tier = "Tier0"
+	Tier1 Tier = "Tier1"
+	Tier2 Tier = "Tier2"
+	Tier3 Tier = "Tier3"
+	Tier4 Tier = "Tier4"
+)
+const (
+	StatusActive      ClusterStatus = "active"
+	StatusMaintenance ClusterStatus = "maintenance"
+	StatusOffline     ClusterStatus = "offline"
+)
+
+type ClusterMetadata struct {
+	Name              string        `json:"name" db:"name"`
+	Status            ClusterStatus `json:"status" db:"status"`
+	StatusReason      string        `json:"status_reason" db:"status_reason"`
+	StatusSince       time.Time     `json:"status_since" db:"status_since"`
+	AllowedTiers      Tiers         `json:"allowed_tiers" db:"allowed_tiers"`
+	UpdatedAt         time.Time     `json:"updated_at" db:"updated_at"`
+	Namespace         string        `json:"namespace" db:"namespace"`
+	Region            string        `json:"region" db:"region"`
+	EMRVirtualCluster string        `json:"emr_virtual_cluster" db:"emr_virtual_cluster"`
+}
+
+func (t *Tiers) Scan(value interface{}) error {
+	if value == nil {
+		return nil
+	}
+
+	switch v := value.(type) {
+	case []byte:
+		return json.Unmarshal(v, t)
+	case string:
+		return json.Unmarshal([]byte(v), t)
```

Comment:
Ah, I see this is using JSON, which might be easier than using a PG-native array data type.  If it's not too much effort, we should look into switching it to that array type though